foodsaving/stores/tests/test_feedback_api.py

```diff
@@ -0,0 +1,48 @@
+from itertools import zip_longest
+
+from dateutil.parser import parse
+from dateutil import rrule
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrulestr
+from django.utils.datetime_safe import datetime
+
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.test import APITestCase
+from foodsaving.groups.factories import GroupFactory
+from foodsaving.stores.factories import StoreFactory, PickupDateFactory, PickupDateSeriesFactory
+from foodsaving.stores.models import PickupDate, Feedback
+from foodsaving.users.factories import UserFactory
+
+
+class TestFeedbackAPI(APITestCase):
+    """
+    This is an unit test for the Feedback API
+    """
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.now = timezone.now()
+        cls.member = UserFactory()
+        cls.group = GroupFactory(members=[cls.member, ])
+        cls.store = StoreFactory(group=cls.group)
+        cls.pickup = PickupDateFactory(store=cls.store)
+        cls.series = PickupDateSeriesFactory(max_collectors=3, store=cls.store)
+        cls.series.update_pickup_dates(start=lambda: cls.now)
+
+    def test_new(self):
+        """
+        User is trying to give a feedback for a pickup
+        that he didn't do (= not part of the collectors)?
+        """
+        user_feedback = Feedback()
```

Comment:
What did you try to do here?
Maybe you could formulate in words first what the test should do, e.g. something like this:

```py
"""
setup:
1. create user
2. create group with the user as member
3. create a store within that group
4. create a pickup within that store that is in the past (to make allow giving feedback)
5. add user to the pickup as collector

test:
1. log user in
2. user gives feedback to pickup
3. make sure that response is valid
4. make sure that feedback has been created
"""
```

This way it's easier for you to have an overview of the tasks that the test needs to do and you don't need to write it directly into code. It's also easier to spot why a test fails: either there were wrong assumptions what the tests needs to to, or it was just not the correct code to do it.

By the way, the steps I described above are for the _correct_ case, which I would always write before the edge cases - it's simply more fulfilling to me to see the code doing the right thing :)